Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -492,116 +502,123 @@ class BocList
+    fakeTableHeadWidthContainer.setAttribute('role', 'none');
     fakeTableHeadWidthContainer.append(fakeTable);
   
-    var fakeTableHeadContainer = $('<div/>').attr({ 'role': 'none', 'class': 'bocListFakeTableHead' });
-    fakeTableHeadContainer.hide();
+    var fakeTableHeadContainer = document.createElement('div');
+    fakeTableHeadContainer.setAttribute('role', 'none');
+    fakeTableHeadContainer.classList.add('bocListFakeTableHead');
+    fakeTableHeadContainer.style.display = 'none';
     fakeTableHeadContainer.append (fakeTableHeadWidthContainer);
   
-    realTableHead.find('*').each(function (this: HTMLElement) { $(this).removeAttr('id').attr({ tabIndex: -1 }).attr({ tabIndex: -1 }); });
+    realTableHead.querySelectorAll('*').forEach(element =>
+    {
+      element.removeAttribute('id');
+      element.setAttribute('tabIndex', '-1');
+    });
   
     scrollableContainer.before(fakeTableHeadContainer);
   
     // sync checkboxes
-    var checkboxes = fakeTableHead.find("th input:checkbox");
-    checkboxes.click(function (this: HTMLElement)
+    var checkboxes = fakeTableHead.querySelectorAll<HTMLInputElement>("th input[type=checkbox]");
+    checkboxes.forEach(checkbox =>
     {
-      var checkName = $(this).attr('name');
-      var checkStatus = $(this).prop('checked');
-      $('input[name="' + checkName + '"]').prop('checked', checkStatus);
+      checkbox.addEventListener('click', (event) =>
+      {
+        var checkName = checkbox.getAttribute('name');
+        var checkStatus = checkbox.checked;
+        document.querySelectorAll('input[name="' + checkName + '"]').forEach(element =>
+        {
+          (element as HTMLInputElement).checked = checkStatus;
+        });
+      })
     });
     // BocList_FixHeaderSize() with timeout needs to be called after setup. This is already taken care of at the call-site.
   }
 
-  private static FixHeaderSize(scrollableContainer: JQuery): void
+  private static FixHeaderSize(scrollableContainer: HTMLElement): void
   {
-    var realTable = scrollableContainer.children('table').first();
-    var realTableWidth = realTable[0].offsetWidth;
-    var previousRealTableWidth = realTable.data("bocListPreviousRealTableWidth");
+    var realTable = scrollableContainer.querySelector<HTMLTableElement>(':scope>table')!;
+    var realTableWidth = Math.floor(LayoutUtility.GetWidth(realTable));
+    var previousRealTableWidth = parseInt(realTable.dataset.bocListPreviousRealTableWidth!);
     if (previousRealTableWidth == realTableWidth)
       return;
-    realTable.data("bocListPreviousRealTableWidth", realTableWidth);
+    realTable.dataset.bocListPreviousRealTableWidth = realTableWidth.toString();
   
-    var realTableHead = realTable.eq(0).find('thead');
-    var realTableHeadRow = realTableHead.children().eq(0);
-    var realTableHeadRowChildren = realTableHeadRow.children();
+    var realTableHead = realTable.querySelector('thead')!;
```

Comment:
please review all usages of querySelector and querySelectorAll for missing usage of `:scope`

@a-ctor what's the mgmt summary on using / not-using `:scope` ?

Reply:
`.children` vs. `.find`
`:scope > ...` vs. `...`

Reply:
Ah, okay, querySelector and querySelectorAll are always rooted. Missed that one :o

Reply:
They are rooted to the element that you apply the `.querySelector` to. So `element.querySelector` will search for the selector only in the descendants of the elements. But for `.children` you want to search for only children of the element you are applying the selector to, which is available in the css selector as `:scope`.